content/developer/addons/theme-hooks.md

```diff
@@ -0,0 +1,138 @@
+
+Here's an example of a themehooks file that sets some config variables, adds locale data for the view, adds a respond button to the discussion page, and overrides a method to add an extra css class to a menu count span.
+
+## Example Themehooks
+
+```php
+<?php
+/**
+ * @author    My Name <name@email.com>
+ * @copyright 2015 (c) My Organizations
+ * @license   http://opensource.org/licenses/MIT MIT
+ */
+
+/**
+ * Adds locale data to the view, and adds a respond button to the discussion page.
+ */
+class MyThemeNameThemeHooks extends Gdn_Plugin {
+
+    /**
+     * Fetches the current locale and sets the data for the theme view.
+     * Render the locale in a smarty template using {$locale}
+     *
+     * @param  Controller $sender The sending controller object.
+     */
+    public function base_render_before($sender) {
+        // Bail out if we're in the dashboard
+        if (inSection('Dashboard')) {
+            return;
+        }
+
+        // Fetch the currently enabled locale (en by default)
+        $locale = Gdn::locale()->current();
+        $sender->setData('locale', $locale);
+    }
+
+    /**
+     * Adds a 'respond' button on a discussion page below the discussion title
+     * that links to the comment form.
+     *
+     * @param DiscussionController $sender The sending object.
+     */
+    public function discussionController_afterDiscussionTitle_handler($sender) {
+        // Ensure the user is signed in.
+        if (Gdn::session()) {
+            echo '<div class="below-discussion-title">'.
+                 '<a class="respond-button Button" href="#Form_Comment">Respond</a>'.
+                 '</div>';
+        }
+    }
+}
+
+/**
+ * Adds a new css class to counts in the side panel filter
+ * menu in a discussion list. Overrides the method in the DiscussionFilterModule
+ * (applications/vanilla/views/modules/discussionfilter.php)
+ *
+ * @param int|null $count The number to include in the count.
+ * @param string $url The url to the popin rel for the count.
+ */
+if (!function_exists('filterCountString')) {
+    function filterCountString($count, $url = '') {
+        $count = countString($count, $url);
+        return $count != '' ? '<span class="a-new-css-class aside">'.$count.'</span>' : '';
+    }
+}
+```
+
+
+## Common Hooks
+
+There are a few common functions for a themehooks file.
+
+### base_render_before()
+
+This runs on every single page load before the render. This includes the Dashboard so be sure to exclude it! This is also a good place to use the `Gdn_Controller` method `setData()` to inject data into your Smarty templates. 
```

Comment:
A link to `Gdn_Controller` and `setData()` would be nice here.

Reply:
I've added these to the Controller documentation and linked to them.